From 67f43a5bf4e59460f4328956f007b39265c28816 Mon Sep 17 00:00:00 2001
From: Tomas Mraz <tomas@openssl.org>
Date: Thu, 29 Aug 2024 18:42:14 +0200
Subject: [PATCH 221/228] endecode_test.c: Avoid running the SM2 tests with
 3.0.0 FIPS provider

Fixes #25326

Reviewed-by: Richard Levitte <levitte@openssl.org>
Reviewed-by: Neil Horman <nhorman@openssl.org>
(Merged from https://github.com/openssl/openssl/pull/25327)

(cherry picked from commit 0b97a5505efa8833bb7b8cabae45894ad6d910a2)
---
 test/endecode_test.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/test/endecode_test.c b/test/endecode_test.c
index 97576cedc7..8e7dc7a9ec 100644
--- a/test/endecode_test.c
+++ b/test/endecode_test.c
@@ -1451,7 +1451,10 @@ int setup_tests(void)
         ADD_TEST_SUITE_LEGACY(ECExplicitTri2G);
 # endif
 # ifndef OPENSSL_NO_SM2
-        ADD_TEST_SUITE(SM2);
+        if (!is_fips_3_0_0) {
+            /* 3.0.0 FIPS provider imports explicit EC params and then fails. */
+            ADD_TEST_SUITE(SM2);
+        }
 # endif
         ADD_TEST_SUITE(ED25519);
         ADD_TEST_SUITE(ED448);
-- 
2.39.5 (Apple Git-154)

